app/src/test/java/org/mozilla/fenix/settings/sharedpreferences/BooleanPreferenceTest.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.sharedpreferences
+
+import android.content.SharedPreferences
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.verify
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class BooleanPreferenceTest {
+
+    private lateinit var sharedPrefs: SharedPreferences
+    private lateinit var editor: SharedPreferences.Editor
+
+    @Before
+    fun setup() {
+        sharedPrefs = mockk(relaxed = true)
+        editor = mockk()
+
+        every { sharedPrefs.edit() } returns editor
+        every { editor.putBoolean(any(), any()) } returns editor
+        every { editor.apply() } just runs
+    }
+
+    @Test
+    fun `getter returns boolean from shared preferences`() {
+        val holder = object : PreferencesHolder {
+            override val preferences = sharedPrefs
+            val test by booleanPreference("test_pref_key")
+        }
+        every { sharedPrefs.getBoolean("test_pref_key", false) } returns true
+
+        assertTrue(holder.test)
+        verify { sharedPrefs.getBoolean("test_pref_key", false) }
+    }
+
+    @Test
+    fun `setter applies boolean to shared preferences`() {
+        val holder = object : PreferencesHolder {
+            override val preferences = sharedPrefs
+            var test by booleanPreference("pref", default = true)
+        }
+        holder.test = false
+
+        verify { editor.putBoolean("pref", false) }
+        verify { editor.apply() }
+    }
+
+    @Test
+    fun `getter uses default value`() {
+        val holderFalse = object : PreferencesHolder {
+            override val preferences = sharedPrefs
+            val test by booleanPreference("test_pref_key", default = false)
+        }
+        holderFalse.test
```

Comment:
nit: perhaps I'm just a bit slow this Monday morning, but it wasn't clear to me that this was testing accessing this variable--it looked like a misplaced line. Consider leaving a comment here or assign it to a variable `readValue = holderFalse.test` and then verify that value? 

Again, maybe just me being slow, if you feel this is clear you can keep as is :)

Reply:
Added comments!